src/jit/lower.cpp

```diff
@@ -2809,7 +2860,7 @@ GenTree* Lowering::LowerNonvirtPinvokeCall(GenTreeCall* call)
 
     InsertPInvokeCallProlog(call);
 
-    if (call->gtCallType != CT_INDIRECT)
+    if ((comp->opts.eeFlags & CORJIT_FLG_PINVOKE_DIRECT_CALLS) == 0 && call->gtCallType != CT_INDIRECT)
```

Comment:
Instead of adding the `CORJIT_FLG_PINVOKE_DIRECT_CALLS` flag, it would be better (consistent with the JIT/EE design) to change `getAddressOfPInvokeFixup` to return CORINFO_CONST_LOOKUP. How many indirections are needed can be then described via `IAT_VALUE/IAT_PVALUE/IAT_PPVALUE`.

(If there is a problem with changing getAddressOfPInvokeFixup, add a new method to the JIT-EE interface that does the right thing.)